doc/source/fmpz_mod_poly_factor.rst

```diff
@@ -182,3 +182,21 @@ Factorisation
     :type:`fmpz_mod_poly_interval_poly_arg_t`.
 
 
+Root Finding
+--------------------------------------------------------------------------------
+
+.. function:: void fmpz_mod_poly_roots(fmpz_mod_poly_factor_t r, const fmpz_mod_poly_t f, int want_mult)
+
+    Fill `r` with factors of the form `x - r_i` where the `r_i` are the distinct roots of a nonzero `f` in `Z/pZ`.
+    It is expected and not checked that `f->p` is prime.
+    If `want_mult` is zero, the exponent `e_i` of the factor `x - r_i` is `1`. Otherwise, it is the largest `e_i` such that `(x-r_i)^e_i` divides `f`.
+    This function throws if `f` is zero, but is otherwise always successful.
+
+.. function:: int fmpz_mod_poly_roots_general(fmpz_mod_poly_factor_t r, const fmpz_mod_poly_t f, int want_mult, const fmpz_factor_t n)
```

Comment:
_general seems like the wrong description here. Would _heuristic accurately describe it? _partial?

Reply:
Hmm, I am not sure heuristic or partial describe anything meaningful. The fmpz_mod_poly_roots finds roots of polynomials over (Z/pZ)[x], where p is prime. I suppose this will be the most common use case.
In order to find roots over (Z/nZ)[x], I need a prime factorization of n, which is what the user has to provide to the function.

Reply:
_factored? I think there is a precedent with some of the arithmetic functions like the euler function. Not sure.